### PR TITLE
Exercise 9

### PR DESCRIPTION
**Pull request by aniekanokpokpo**

### CODE REVIEW

**Comment by aniekanokpokpo** [comment]
There is some issues running the application on Heroku. For some reason, it's unable to connect to Atlas mongo. I am able to ping the database and see the databases and collections, however when accessing the page, get error with code 8000, stating AtlasError "Authentication failed"

File: todo_app/mongo_repository.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        items = []
+        for doc in self.todo_collection.find():
+            items.append(Item.from_response(doc))
+
+        return items
+
+    def move_to_done(self, item_id):
+        saved = self.todo_collection.find_one_and_update({'_id': ObjectId(item_id)},
+                                                         {'$set': {'status': 'COMPLETED'}},
+                                                         return_document=ReturnDocument.AFTER)
```
Comment (BenRamchandani) [inline todo_app/mongo_repository.py]: You don't do anything with the returned document

File: README.md [hunk 2]
```diff
@@ -60,3 +60,7 @@ Now visit [`http://localhost:5000/`](http://localhost:5000/) in your web browser
 #### how to run tests
 - ensure chromedriver is downloaded and added to system path (https://chromedriver.chromium.org/downloads) 
 - run `poetry run pytest` from the command line
+
+### Deployment
+The application is deployed to heroku when a new commit/merge is made to the `master` branch
```
Comment (BenRamchandani) [inline README.md:67]: Your README still mentions `SECRET_KEY` which you don't use any more. Would be nice to briefly mention setting up the mongo connection variables in .env instead.

File: todo_app/item.py [hunk 3]
```diff
@@ -14,13 +14,12 @@ def __init__(self, item_id, tittle, status, desc, last_changed=date.today()):
...
         r_title = response['name']
-        r_status = Status.COMPLETED if config.DONE_LIST_ID == response['idList'] else Status.NOT_STARTED
-        r_id = response['id']
+        r_status = Status[response['status']]
+        r_id = response['_id']
         r_desc = response['desc']
-        datestr = response['dateLastActivity'].split('T')[0]
-        r_last_changed = datetime.fromisoformat(datestr).date()
+        r_last_changed = response['dateLastActivity'].date()
+        # r_last_changed = datetime.fromisoformat(datestr).date()
```
Comment (BenRamchandani) [inline todo_app/item.py]: Delete the commented out code

File: tests/test_selenuim.py [hunk 4]
```diff
@@ -1,39 +1,28 @@
...
-    thread = Thread(target=lambda: application.run(use_reloader=False))
-    thread.daemon = True
-    thread.start()
-    yield app
-    # Tear Down
-    thread.join(1)
-    trello_boards.delete_board(board_id)
+    with mongomock.patch(servers=('mongodb://example:27017/',)):
+        file_path = find_dotenv('../.env.test')
+        load_dotenv(file_path, override=True)
```
Comment (BenRamchandani) [inline tests/test_selenuim.py:15]: It should be fine to let your selenium test use the real `.env` and mongo database, though mocking it out also works

**Review by BenRamchandani — APPROVED** [review]
Mongo integration looks good.

A couple of things on your tests that aren't related to this exercise but might be worth looking at:
* `with webdriver.Chrome('./chromedriver', options=opts) as driver:` assumes `chromedriver` exists in the current directory - better to do `with webdriver.Chrome('chromedriver', options=opts) as driver:` so it works as long as it's on your path. Check this doesn't break docker tests though.
* I haven't tried it, but does your Travis pipeline fail if one of your non-selenium tests fails? I would expect it to just carry on unless you `set -e` at the start of the travis_test script

File: todo_app/mongo_repository.py (hunk 1 above)
Comment (aniekanokpokpo) [reply todo_app/mongo_repository.py]: removed

File: todo_app/item.py (hunk 3 above)
Comment (aniekanokpokpo) [reply todo_app/item.py]: removed

File: README.md (hunk 2 above)
Comment (aniekanokpokpo) [reply README.md:67]: updated

**Pull request merged**